lisa/operating_system.py

```diff
@@ -131,16 +170,47 @@ def _get_detect_string(cls, node: Any) -> Iterable[str]:
         cmd_result = typed_node.execute(cmd="cat /etc/SuSE-release", no_error_log=True)
         yield get_matched_str(cmd_result.stdout, cls.__suse_release_pattern)
 
+    def _get_os_version(self) -> OsVersion:
+        return self._os_version
+
 
 class Windows(OperatingSystem):
     def __init__(self, node: Any) -> None:
         super().__init__(node, is_posix=False)
+        self._get_os_version
+
+    def _get_os_version(self) -> OsVersion:
+        self._os_version.vendor = "Microsoft Corporation"
+        cmd_result = self._node.execute(
+            cmd="systeminfo | findstr /B /C:'OS Version'",
+            no_error_log=True,
+        )
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            for row in cmd_result.stdout.split("\n"):
+                if ": " in row:
+                    key, value = row.split(": ")
+                    if "OS Version" in key:
+                        self._os_version.release = value.strip()
+        return self._os_version
 
 
 class Posix(OperatingSystem, BaseClassMixin):
+
+    __url_pattern = re.compile(
```

Comment:
For each regex, please provide the examples to parse. So if someone update the regex in future, it can be tested easily. And it's also helpful on explaining what this regexp do.

Reply:
Sure, not using this regex for now.